src/App.tsx

```diff
@@ -8,37 +8,91 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { getUsers } from './services/users';
+import { getUserPosts } from './services/userPosts';
+import { User } from './types/User';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  const handleButtonClick = (post: Post) => {
+    if (selectedPost?.id === post.id) {
+      setSelectedPost(null);
+    } else {
+      setSelectedPost(post);
+    }
+  };
+
+  useEffect(() => {
+    getUsers()
+      .then(setUsers);
+  }, []);
+
+  useEffect(() => {
+    if (selectedUser) {
+      setLoading(true);
+
+      getUserPosts(selectedUser.id)
+        .then(setUserPosts)
+        .catch(() => setErrorMessage('Something went wrong!'))
+        .finally(() => setLoading(false));
+    }
+  }, [selectedUser]);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {loading && (
+                  <Loader />
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {!loading && errorMessage && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    {errorMessage}
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {!loading && !errorMessage
+                && userPosts.length === 0 && selectedUser && (
```

Comment:
pls move long ternaries to separate const with consistent names and use them here